### PR TITLE
Optimalbit/bull add failedReason to Job type

### PR DESCRIPTION
**Pull request by reediculous456**
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/OptimalBits/bull/blob/master/lib/job.js#L239
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by levibostian** [comment]
To confirm, [the proposed addition has been deployed in the latest version of bull](https://github.com/OptimalBits/bull/blob/v3.13.0/lib/job.js#L239). It's not just on the `develop` branch of the repo. 

Looks good to me.

File: types/bull/index.d.ts [hunk 1]
```diff
@@ -171,6 +171,8 @@ declare namespace Bull {
 
     returnvalue: any;
 
+    failedReason: string;
```
Comment (bondz) [inline types/bull/index.d.ts:174]: It can still be undefined if the job doesn't fail.

**Comment by reediculous456** [comment]
> To confirm, [the proposed addition has been deployed in the latest version of bull](https://github.com/OptimalBits/bull/blob/v3.13.0/lib/job.js#L239). It's not just on the `develop` branch of the repo.
> 
> Looks good to me.

My bad, linked the wrong branch. I have fixed that in the issue text

File: types/bull/index.d.ts (hunk 1 above)
Comment (reediculous456) [reply types/bull/index.d.ts:174]: So should it be changed to be this?
```suggestion
    failedReason: string | undefined;
```

Comment (bondz) [reply types/bull/index.d.ts:174]: Yes, or
```suggestion
    failedReason?: string;
```

Comment (reediculous456) [reply types/bull/index.d.ts]: I always forget about the ? optional property. It looks a lot cleaner 😆

**Review by bondz — APPROVED** [review]
👍

**Pull request merged**